src/Numerical/DiscontinuousGalerkin/Actions/FluxCommunication.hpp

```diff
@@ -0,0 +1,321 @@
+      const auto& local_value = local_data.at(direction);
+      const auto& neighbor_value = neighbor_data_it->second;
+
+      // This needs to be fixed for hp-adaptivity to handle
+      // projections correctly
+
+      const auto grid_normal = unnormalized_grid_normal(
+          extents.slice_away(dimension), element_map, direction);
+      // Using this instead of auto prevents incomprehensible errors
+      // if the return type of compute_flux is wrong.
+      using FluxType = db::item_type<add_tag_prefix<
+          Tags::Flux, variables_tag, tmpl::size_t<volume_dim>, Frame::Grid>>;
+      const FluxType local_boundary_flux =
+          System::compute_flux::apply(local_value);
+      const FluxType neighbor_boundary_flux =
+          System::compute_flux::apply(neighbor_value);
+      using normal_flux_tag =
+          add_tag_prefix<Tags::NormalDotFlux, variables_tag>;
+      db::item_type<normal_flux_tag> local_normal_flux(
+          local_boundary_flux.number_of_grid_points(), 0.);
+      db::item_type<normal_flux_tag> neighbor_normal_flux(
+          local_boundary_flux.number_of_grid_points(), 0.);
+      tmpl::for_each<typename variables_tag::tags_list>([
+        &local_normal_flux, &neighbor_normal_flux, &local_boundary_flux,
+        &neighbor_boundary_flux, &grid_normal
+      ](auto tag) noexcept {
+        using Tag = tmpl::type_from<decltype(tag)>;
+        using flux_tag = Tags::Flux<Tag, tmpl::size_t<2>, Frame::Grid>;
+        using normal_flux_tag = Tags::NormalDotFlux<Tag>;
+
+        const auto& local_bf = get<flux_tag>(local_boundary_flux);
+        auto& local_nf = get<normal_flux_tag>(local_normal_flux);
+        const auto& neighbor_bf = get<flux_tag>(neighbor_boundary_flux);
+        auto& neighbor_nf = get<normal_flux_tag>(neighbor_normal_flux);
+        for (auto flux_it = local_bf.begin(); flux_it != local_bf.end();
+             ++flux_it) {
+          const auto flux_index =
+              local_bf.get_tensor_index(flux_it);
+          const size_t contract_index = flux_index[0];
+          const auto other_indices =
+              all_but_specified_element_of<0>(flux_index);
+          local_nf.get(other_indices) +=
+              grid_normal.get(contract_index) * local_bf.get(flux_index);
+          neighbor_nf.get(other_indices) +=
+              grid_normal.get(contract_index) * neighbor_bf.get(flux_index);
+        }
+      });
+
+      const tuples::TaggedTuple<variables_tag, normal_flux_tag> self_data(
+          local_value, std::move(local_normal_flux));
+      const tuples::TaggedTuple<variables_tag, normal_flux_tag> neighbor_data(
+          neighbor_value, std::move(neighbor_normal_flux));
+
+      auto numerical_flux = apply_flux(
+          flux_computer, self_data, neighbor_data,
+          typename std::decay_t<decltype(flux_computer)>::argument_tags{});
+
+      DataVector magnitude_of_grid_normal = magnitude(grid_normal);
```

Comment:
This will need a metric for non-Euclidean systems

Reply:
adding a comment is fine

Reply:
'GH modification needed' wouldn't be correct since the modification would be needed for static background systems aswell (ScalarWaveCurved and RelativisticEuler).

Reply:
Then suggest something that you'll be able to find...

Reply:
I mean anything will do... Even 'GR modification needed' is fine.